src/celeritas/optical/surface/VolumeSurfaceSelector.hh

```diff
@@ -0,0 +1,128 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/surface/VolumeSurfaceSelector.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Assert.hh"
+#include "corecel/Macros.hh"
+#include "geocel/SurfaceData.hh"
+#include "geocel/VolumeSurfaceView.hh"
+#include "celeritas/geo/GeoTrackView.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Retrieve the surface ID between two volume instances based on Geant4's
+ * interface / boundary priority.
+ *
+ * Given (old, new) physical volumes P0, P1 corresponding to logical volumes
+ * L0, L1
+ * - Ordered (P0, P1) border surface
+ * - Skin surface of L1 if it's the daughter of L0
```

Comment:
Let's add a note for now that we haven't implemented this case, so this differs a bit from Geant4's heuristic.